chore(components): use TypeScript 3.7 features (#168)
carbon-design-system/carbon-custom-elements: src/coding-conventions.md

If you get TypeScript "may be null" errors, think twice to see if there is such edge case:

> - If some other portion of your code ensures the `null` condition won't happen and somebody else won't likely break that, use non-null assertion operator (`!`) - But don't do that blindly.

Reviewer: ```suggestion
- If some other portion of your code ensures the `null` condition won't happen and nothing else is likely to break it, use the non-null assertion operator (`!`) - But don't blindly do so.
```

> - Otherwise, add code to do `null` check to do one of the following:

Reviewer: ```suggestion
- Otherwise, add code to perform a `null` check by doing one of the following:
```

>   - Throw an exception that well explains why the `null` value won't be acceptable and (if applicable) what mistake may cause that wrong condition

Reviewer: ```suggestion
  - Throw an exception that explains why the `null` value won't be acceptable and (if applicable) what mistake may cause that wrong condition
```
